bot/kodiak/evaluation.py

```diff
@@ -750,11 +757,44 @@ async def set_status(msg: str, markdown_content: Optional[str] = None) -> None:
     if (config.merge.prioritize_ready_to_merge and ready_to_merge) or merging:
         merge_args = get_merge_body(config, pull_request)
         await set_status("⛴ attempting to merge PR (merging)")
-        await api.merge(
-            merge_method=merge_args.merge_method,
-            commit_title=merge_args.commit_title,
-            commit_message=merge_args.commit_message,
-        )
+        try:
+            await api.merge(
+                merge_method=merge_args.merge_method,
+                commit_title=merge_args.commit_title,
+                commit_message=merge_args.commit_message,
+            )
+        # if we encounter an internal server error (status code 500), it is
+        # _not_ safe to retry. Instead we mark the pull request as unmergable
+        # and require a user to re-enable Kodiak on the pull request.
+        except GitHubApiInternalServerError:
```

Comment:
I know we have some retry logic for API exceptions that runs outside of `mergeable()`, but I think this is the proper place to handle this error. I think we should also try to move that retry logic in to the function as well.

We can easily provide thorough test coverage of any logic inside of `mergable()`.